src/core/dev/hw_queue_rx.cpp

```diff
@@ -189,6 +191,25 @@ bool hw_queue_rx::prepare_doca_rxq()
         return false;
     }
 
+#if defined(DEFINED_NGINX) || defined(DEFINED_ENVOY)
+    /*
+     * For some scenario with forking usage we may want to distribute CQs across multiple
+     * CPUs to improve CPS in case of multiple processes.
+     */
+    if (safe_mce_sys().app.distribute_cq_interrupts && g_p_app->get_worker_id() >= 0 &&
+        m_p_ib_ctx_handler->is_notification_affinity_supported()) {
+        uint32_t core = g_p_app->get_worker_id() % std::thread::hardware_concurrency();
```

Comment:
from the hardware_concurrency() description:
```
Returns the number of concurrent threads supported by the implementation. The value should be considered only a hint.
```
Is it reliable approach? What if available CPUs have non-continuous numbering? What if number of completion channels (IRQs) is lower than number of CPUs?

Reply:
You are correct. This implementation is not good because CPU can be higher than comp channels. We need to ask from DOCA to give us the number of available channels.
I will change this section to use some temporary constant comp ch number just to test functionality.